### PR TITLE
Update IHostingEnvironment to IWebHostEnvironment

### PR DESCRIPTION
**Pull request by ardalis**
IHostingEnvironment is obsolete.

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@ardalis  much appreciated.
Fixes #21154

File: aspnetcore/mvc/views/tag-helpers/built-in/environment-tag-helper.md [hunk 1]
```diff
@@ -22,13 +22,13 @@ For an overview of Tag Helpers, see <xref:mvc/views/tag-helpers/intro>.
 
 `names` accepts a single hosting environment name or a comma-separated list of hosting environment names that trigger the rendering of the enclosed content.
 
-Environment values are compared to the current value returned by [IHostingEnvironment.EnvironmentName](xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment.EnvironmentName*). The comparison ignores case.
+Environment values are compared to the current value returned by [IWebHostEnvironment.EnvironmentName](xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment.EnvironmentName*). The comparison ignores case.
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/built-in/environment-tag-helper.md:25]:
```suggestion
Environment values are compared to the current value returned by <xref:Microsoft.AspNetCore.Hosting.EnvironmentName>. The comparison ignores case.
```

**Comment by Rick-Anderson** [comment]
Line 25: [Warning] Cross reference not found: 'Microsoft.AspNetCore.Hosting.IWebHostEnvironment.EnvironmentName*'.
Line 43: [Warning] Cross reference not found: 'Microsoft.AspNetCore.Hosting.IWebHostEnvironment.EnvironmentName*'.

**Comment by ardalis** [comment]
Hmm, it works in my local Visual Studio. Is it just missing from the API docs?

**Comment by Rick-Anderson** [comment]
> Hmm, it works in my local Visual Studio. Is it just missing from the API docs?

I'll look into it.
@scottaddie  any idea what's wrong here?

**Comment by serpent5** [comment]
@Rick-Anderson Is it because `EnvironmentName` is _declared_ in `IHostEnvironment` and only _inherited_ into `IWebHostEnvironment`?

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
@ardalis  thanks for the PR. I'll open an issue of the missing API.

**Comment by Rick-Anderson** [comment]
see https://github.com/dotnet/AspNetApiDocs/issues/194

**Pull request merged**